Database/TestDatabaseCache.php

```diff
@@ -0,0 +1,152 @@
+
+use Symfony\Component\DependencyInjection\Container;
+use Doctrine\Common\DataFixtures\Executor\AbstractExecutor;
+
+class TestDatabaseCache
+{
+    private $container;
+
+    public function __construct(Container $container)
+    {
+        $this->container = $container;
+    }
+
+    public function getCachedExecutor(TestDatabasePreparator $dbPreparator, array $classNames)
+    {
+        $backup = $this->buildCacheFilePath($dbPreparator->getMetaDatas(), $classNames);
+        if ($this->isBackupUpToDate($classNames, $backup)) {
+            $om = $dbPreparator->getObjectManager();
+            $om->flush();
+            $om->clear();
+
+            $executor = $dbPreparator->getExecutorWithReferenceRepository();
+            $executor->getReferenceRepository()->load($backup);
+
+            copy($backup, $this->getSQLiteName($om->getConnection()->getParams()));
+
+            return $executor;
+        }
+    }
+
+    public function storeToCache(AbstractExecutor $executor, TestDatabasePreparator $dbPreparator, array $classNames)
+    {
+        $backup = $this->buildCacheFilePath($dbPreparator->getMetaDatas(), $classNames);
+        $executor->getReferenceRepository()->save($backup);
+        copy($this->getSQLiteName($dbPreparator->getObjectManager()->getConnection()->getParams()), $backup);
+    }
+
+    /**
+     * This function finds the time when the data blocks of a class definition
+     * file were being written to, that is, the time when the content of the
+     * file was changed.
+     *
+     * @param string $class
+     *            The fully qualified class name of the fixture class to
+     *            check modification date on.
+     *
+     * @return \DateTime|null
+     */
+    protected function getClassLastModified($class)
+    {
+        $lastModifiedDateTime = null;
+
+        $reflClass = new \ReflectionClass($class);
+        $classFileName = $reflClass->getFileName();
+
+        if (file_exists($classFileName)) {
+            $lastModifiedDateTime = new \DateTime();
+            $lastModifiedDateTime->setTimestamp(filemtime($classFileName));
+        }
+
+        return $lastModifiedDateTime;
+    }
+
+    /**
+     * Determine if the Fixtures that define a database backup have been
+     * modified since the backup was made.
+     *
+     * @param array $classNames
+     *            The fixture classnames to check
+     * @param string $backup
+     *            The fixture backup SQLite database file path
+     *
+     * @return bool TRUE if the backup was made since the modifications to the
+     *         fixtures; FALSE otherwise
+     */
+    private function isBackupUpToDate(array $classNames, $backup)
+    {
+        if(!file_exists($backup) || !file_exists($backup.'.ser')) {
+            return false;
+        }
+
+        $backupLastModifiedDateTime = new \DateTime();
+        $backupLastModifiedDateTime->setTimestamp(filemtime($backup));
+
+        foreach ($classNames as &$className) {
```

Comment:
why using references here ? this is a bad idea, especially given you pass it to a function expecting a non-reference just after, causing a reference mismatch (see http://jpauli.github.io/2014/06/27/references-mismatch.html for the explanation)